test/integration/replicaset/replicaset_test.go

```diff
@@ -930,3 +931,166 @@ func TestFullyLabeledReplicas(t *testing.T) {
 		t.Fatalf("Failed to verify only one pod is fully labeled: %v", err)
 	}
 }
+
+func getOrphanOptions() *metav1.DeleteOptions {
+	var trueVar = true
+	return &metav1.DeleteOptions{OrphanDependents: &trueVar}
+}
+
+func TestReplicaSetsExtensionsV1beta1DefaultGCPolicy(t *testing.T) {
+	s, closeFn, rm, informers, c := rmSetup(t)
+	defer closeFn()
+	ns := framework.CreateTestingNamespace("test-default-gc-extensions", s, t)
+	defer framework.DeleteTestingNamespace(ns, s, t)
+	stopCh := runControllerAndInformers(t, rm, informers, 0)
+	defer close(stopCh)
+
+	rs := newRS("rs", ns.Name, 2)
+	fakeFinalizer := "kube.io/dummy-finalizer"
+	rs.Finalizers = []string{fakeFinalizer}
+	rss, _ := createRSsPods(t, c, []*v1beta1.ReplicaSet{rs}, []*v1.Pod{})
+	rs = rss[0]
+	waitRSStable(t, c, rs)
+
+	// Verify RS creates 2 pods
+	podClient := c.CoreV1().Pods(ns.Name)
+	pods := getPods(t, podClient, labelMap())
+	if len(pods.Items) != 2 {
+		t.Fatalf("len(pods) = %d, want 2", len(pods.Items))
+	}
+
+	rsClient := c.ExtensionsV1beta1().ReplicaSets(ns.Name)
+	err := rsClient.Delete(rs.Name, getOrphanOptions())
+	if err != nil {
+		t.Fatalf("Failed to delete rs: %v", err)
+	}
+
+	// Verify orphan finalizer has been added
+	if err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+		newRS, err := rsClient.Get(rs.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return slice.ContainsString(newRS.Finalizers, metav1.FinalizerOrphanDependents, nil), nil
+	}); err != nil {
+		t.Fatalf("Failed to verify orphan finalizer is added: %v", err)
+	}
+
+	updateRS(t, rsClient, rs.Name, func(rs *v1beta1.ReplicaSet) {
+		var finalizers []string
+		// remove fakeFinalizer
+		for _, finalizer := range rs.Finalizers {
+			if finalizer != fakeFinalizer {
+				finalizers = append(finalizers, finalizer)
+			}
+		}
+		rs.Finalizers = finalizers
+	})
+
+	rsClient.Delete(rs.Name, nil)
+}
+
+// TODO(dixudx): enable this test when replicaset controller is bumped to apps/v1
+//func TestReplicaSetsAppsV1DefaultGCPolicy(t *testing.T) {
```

Comment:
Actually, the rs version used in rs controller shouldn't matter. 

What we want to test is something like:
- `c.ExtensionsV1().ReplicaSets(name).Delete(name, nil)` (which uses default GC policy) will orphan all dependents. 
- `c.AppsV1().ReplicaSets(name).Delete(name, nil)` (which uses default GC policy) will delete all dependents. 

You can create whatever version of RS and then tests both endpoints. Delete option should be `nil` so that default GC policy is used.